### ISSUE TITLE
kurento-client build issue

### ISSUE THREAD

**Issue by sm2017** [Kurento/bugtracker#386]
I clone kurento-tutorial-node and want to run kurento-hello-world , when I run `yarn install` or `npm i` , I have an issue with kurento-client

 kurento-client has utf-8-validate@1.2.2 dependency , Please upgrade dependencies , I think it is not compatible with node v10.15.3

Here is errors

```
error /home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@10.16.0 | linux | x64
gyp http GET https://nodejs.org/download/release/v10.16.0/node-v10.16.0-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v10.16.0/node-v10.16.0-headers.tar.gz
gyp http GET https://nodejs.org/download/release/v10.16.0/SHASUMS256.txt
gyp http 200 https://nodejs.org/download/release/v10.16.0/SHASUMS256.txt
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/root/.node-gyp/10.16.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/root/.node-gyp/10.16.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/root/.node-gyp/10.16.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory `/home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate/build'
  CXX(target) Release/obj.target/validation/src/validation.o
In file included from ../node_modules/nan/nan.h:190:0,
                 from ../src/validation.cc:15:
../node_modules/nan/nan_maybe_43_inl.h: In function â€˜Nan::Maybe<bool> Nan::ForceSet(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Value>, v8::PropertyAttribute)â€™:
../node_modules/nan/nan_maybe_43_inl.h:88:15: error: â€˜class v8::Objectâ€™ has no member named â€˜ForceSetâ€™
   return obj->ForceSet(GetCurrentContext(), key, value, attribs);
               ^
In file included from ../src/validation.cc:15:0:
../node_modules/nan/nan.h: In function â€˜v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)â€™:
../node_modules/nan/nan.h:817:60: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/node.h:176): Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, func, argc, argv);
                                                            ^
../node_modules/nan/nan.h: In function â€˜v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)â€™:
../node_modules/nan/nan.h:831:62: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/node.h:169): Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, symbol, argc, argv);
                                                              ^
../node_modules/nan/nan.h: In function â€˜v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)â€™:
../node_modules/nan/nan.h:845:62: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/node.h:162): Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
         v8::Isolate::GetCurrent(), target, method, argc, argv);
                                                              ^
../node_modules/nan/nan.h: In member function â€˜v8::Local<v8::Value> Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int, v8::Local<v8::Value>*) constâ€™:
../node_modules/nan/nan.h:1453:5: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/node.h:176): Use MakeCallback(..., async_context) [-Wdeprecated-declarations]
     ));
     ^
../src/validation.cc: In static member function â€˜static Nan::NAN_METHOD_RETURN_TYPE Validation::IsValidUTF8(Nan::NAN_METHOD_ARGS_TYPE)â€™:
../src/validation.cc:131:50: warning: â€˜v8::Local<v8::Object> v8::Value::ToObject() constâ€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/v8.h:10046): Use maybe version [-Wdeprecated-declarations]
     Local<Object> buffer_obj = info[0]->ToObject();
                                                  ^
make: *** [Release/obj.target/validation/src/validation.o] Error 1
make: Leaving directory `/home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:198:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Linux 3.10.0-957.21.3.el7.x86_64
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate
gyp ERR! node -v v10.16.0
gyp ERR! node-gyp -v v3.8.0


[root@core kurento-hello-world]# wn args   '--generator-output',
c
-bash: wn: command not found
[root@core kurento-hello-world]# gyp info spawn args   'build',
9
-bash: gyp: command not found
[root@core kurento-hello-world]# gyp info spawn args   '-Goutput_dir=.' ]
l
-bash: gyp: command not found
[root@core kurento-hello-world]# gyp info spawn make
/
-bash: gyp: command not found
[root@core kurento-hello-world]# gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
-bash: gyp: command not found
[root@core kurento-hello-world]# make: Entering directory `/home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate/build'
>   CXX(target) Release/obj.target/validation/src/validation.o
> In file included from ../node_modules/nan/nan.h:190:0,
>                  from ../src/validation.cc:15:
> ../node_modules/nan/nan_maybe_43_inl.h: In function â€˜Nan::Maybe<bool> Nan::ForceSet(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Value>, v8::PropertyAttribute)â€™:
> ../node_modules/nan/nan_maybe_43_inl.h:88:15: error: â€˜class v8::Objectâ€™ has no member named â€˜ForceSetâ€™
>    return obj->ForceSet(GetCurrentContext(), key, value, attribs);
>                ^
> In file included from ../src/validation.cc:15:0:
> ../node_modules/nan/nan.h: In function â€˜v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)â€™:
> ../node_modules/nan/nan.h:817:60: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/node.h:176): Use Maklback(..., async_context) [-Wdeprecated-declarations]
>          v8::Isolate::GetCurrent(), target, func, argc, argv);
>                                                             ^
> ../node_modules/nan/nan.h: In function â€˜v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)â€™:
> ../node_modules/nan/nan.h:831:62: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/node.h:169): Use MakeCack(..., async_context) [-Wdeprecated-declarations]
>          v8::Isolate::GetCurrent(), target, symbol, argc, argv);
>                                                               ^
> ../node_modules/nan/nan.h: In function â€˜v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)â€™:
> ../node_modules/nan/nan.h:845:62: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/node.h:162): Use MakeCallback(..sync_context) [-Wdeprecated-declarations]
>          v8::Isolate::GetCurrent(), target, method, argc, argv);
>                                                               ^
> ../node_modules/nan/nan.h: In member function â€˜v8::Local<v8::Value> Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int, v8::Local<v8::Value>*) constâ€™:
> ../node_modules/nan/nan.h:1453:5: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/node.h:176): Use Maklback(..., async_context) [-Wdeprecated-declarations]
>      ));
>      ^
> ../src/validation.cc: In static member function â€˜static Nan::NAN_METHOD_RETURN_TYPE Validation::IsValidUTF8(Nan::NAN_METHOD_ARGS_TYPE)â€™:
> ../src/validation.cc:131:50: warning: â€˜v8::Local<v8::Object> v8::Value::ToObject() constâ€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/v8.h:10046): Use maybe version [-Wdeprecated-declarations]
>      Local<Object> buffer_obj = info[0]->ToObject();
>                                                   ^
> make: *** [Release/obj.target/validation/src/validation.o] Error 1
> make: Leaving directory `/home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate/build'
> gyp ERR! build error 
> gyp ERR! stack Error: `make` failed with exit code: 2
> gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
> gyp ERR! stack     at ChildProcess.emit (events.js:198:13)
> gyp ERR! stack     at Process.ChildProc
> gyp ERR! System Linux 3.10.0-9
> gyp ERR! command "/usr/bin/node" "/usr/l
> gyp ERR! cwd /home/
> gyp ERR! node -v v10.16.0
> gyp ERR! node-gyp -v v3.8.0
>  System Linux 3.10.0-957.21.3.el7.x86_64
> gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
> gyp ERR! cwd /home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate
> gyp ERR! node -v v10.16.0
> gyp ERR! node-gyp -v v3.8.0
> 
> 
> [root@core kurento-hello-world]# wn args   '--generator-output',
> c
> -bash: wn: command not found
> [root@core kurento-hello-world]# gyp info spawn args   'build',
> 9
> -bash: gyp: command not found
> [root@core kurento-hello-world]# gyp info spawn args   '-Goutput_dir=.' ]
> l
> -bash: gyp: command not found
> [root@core kurento-hello-world]# gyp info spawn make
> /
> -bash: gyp: command not found
> [root@core kurento-hello-world]# gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
> -bash: gyp: command not found
> [root@core kurento-hello-world]# make: Entering directory `/home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate/build'
-bash: command substitution: line 1: unexpected EOF while looking for matching `''
-bash: command substitution: line 32: syntax error: unexpected end of file
-bash: make:: command not found
[root@core kurento-hello-world]# >   CXX(target) Release/obj.target/validation/src/validation.o
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# > In file included from ../node_modules/nan/nan.h:190:0,
[root@core kurento-hello-world]# >                  from ../src/validation.cc:15:
-bash: ../src/validation.cc:15:: No such file or directory
[root@core kurento-hello-world]# > ../node_modules/nan/nan_maybe_43_inl.h: In function â€˜Nan::Maybe<bool> Nan::ForceSet(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Value>, v8::PropertyAttribute)â€™:
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# > ../node_modules/nan/nan_maybe_43_inl.h:88:15: error: â€˜class v8::Objectâ€™ has no member named â€˜ForceSetâ€™
-bash: ../node_modules/nan/nan_maybe_43_inl.h:88:15:: No such file or directory
[root@core kurento-hello-world]# >    return obj->ForceSet(GetCurrentContext(), key, value, attribs);
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >                ^
[root@core kurento-hello-world]# > In file included from ../src/validation.cc:15:0:
[root@core kurento-hello-world]# > ../node_modules/nan/nan.h: In function â€˜v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)â€™:
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# > ../node_modules/nan/nan.h:817:60: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/ude/node/node.h:176): Use Maklback(..., async_context) [-Wdeprecated-declarations]
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >          v8::Isolate::GetCurrent(), target, func, argc, argv);
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >                                                             ^
[root@core kurento-hello-world]# > ../node_modules/nan/nan.h: In function â€˜v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)â€™:
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# > ../node_modules/nan/nan.h:831:62: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/ine/node/node.h:169): Use MakeCack(..., async_context) [-Wdeprecated-declarations]
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >          v8::Isolate::GetCurrent(), target, symbol, argc, argv);
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >                                                               ^
[root@core kurento-hello-world]# > ../node_modules/nan/nan.h: In function â€˜v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)â€™:
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# > ../node_modules/nan/nan.h:845:62: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/nodee.h:162): Use MakeCallback(..sync_context) [-Wdeprecated-declarations]
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >          v8::Isolate::GetCurrent(), target, method, argc, argv);
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >                                                               ^
[root@core kurento-hello-world]# > ../node_modules/nan/nan.h: In member function â€˜v8::Local<v8::Value> Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int, v8::Local<v8::Value>*) constâ€™:
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# > ../node_modules/nan/nan.h:1453:5: warning: â€˜v8::Local<v8::Value> node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*)â€™ is deprecated (declared at /root/.node-gyp/10.16.0/ude/node/node.h:176): Use Maklback(..., async_context) [-Wdeprecated-declarations]
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >      ));
-bash: syntax error near unexpected token `)'
[root@core kurento-hello-world]# >      ^
[root@core kurento-hello-world]# > ../src/validation.cc: In static member function â€˜static Nan::NAN_METHOD_RETURN_TYPE Validation::IsValidUTF8(Nan::NAN_METHOD_ARGS_TYPE)â€™:
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# > ../src/validation.cc:131:50: warning: â€˜v8::Local<v8::Object> v8::Value::ToObject() constâ€™ is deprecated (declared at /root/.node-gyp/10.16.0/include/node/v8.h:10046): Use maybe version [-Wdeprecated-declarations]
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >      Local<Object> buffer_obj = info[0]->ToObject();
-bash: syntax error near unexpected token `('
[root@core kurento-hello-world]# >                                                   ^
[root@core kurento-hello-world]# > make: *** [Release/obj.target/validation/src/validation.o] Error 1
-bash: ^: command not found
[root@core kurento-hello-world]# > make: Leaving directory `/home/test/kurento-tutorial-node/kurento-hello-world/node_modules/utf-8-validate/build'
> > gyp ERR! build error 
> > gyp ERR! stack Error: `make` failed with exit code: 2
> > gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
> > gyp ERR! stack     at ChildProcess.emit (events.js:198:13)
> > gyp ERR! stack     at Process.ChildProc
> > gyp ERR! System Linux 3.10.0-9
> > gyp ERR! command "/usr/bin/node" "/usr/l
> > gyp ERR! cwd /home/
> > gyp ERR! node -v v10.16.0
> > gyp ERR! node-gyp -v v3.8.0
> 
```

**Comment by sm2017** [comment]
You must upgrade https://github.com/Kurento/kurento-jsonrpc-js/blob/master/package.json#L64-L66

**Comment by sm2017** [comment]
@j1elo , @agonza1 I tested in windows 10 and centos 7 , same error on both environments

When but I can install latest version of utf-8-validate with no issue

**Comment by sm2017** [comment]
@j1elo , @agonza1 can you please reply

**Comment by j1elo** [comment]
I just merged https://github.com/Kurento/kurento-jsonrpc-js/pull/4 which sould fix this issue

**Comment by sm2017** [comment]
@j1elo `"ws": "^1.1.1"` has old utf-8-validate as dependency, you must upgrade it too

**Comment by j1elo** [comment]
Fixed in commits Kurento/kurento-jsonrpc-js@2f9c17b, Kurento/kurento-jsonrpc-js@4a633fd, after merging the changes suggested in several PRs.
Thanks!
